c++/src/kj/compat/tls.c++

```diff
@@ -248,16 +243,26 @@ public:
   }
 
   kj::Promise<size_t> tryRead(void* buffer, size_t minBytes, size_t maxBytes) override {
-    return tryReadInternal(buffer, minBytes, maxBytes, 0);
+    size_t alreadyDone = 0;
+    while (true) {
+      auto n = co_await sslCall([this,buffer,maxBytes]() { return SSL_read(ssl, buffer, maxBytes); });
+      if (n >= minBytes || n == 0) {
+        co_return alreadyDone + n;
+      } else {
+        buffer = reinterpret_cast<byte*>(buffer) + n;
+        minBytes -= n;
+        maxBytes -= n;
+        alreadyDone += n;
+      }
+    }
   }
 
   Promise<void> write(const void* buffer, size_t size) override {
-    return writeInternal(kj::arrayPtr(reinterpret_cast<const byte*>(buffer), size), nullptr);
+    return writeInternal({kj::arrayPtr(reinterpret_cast<const byte*>(buffer), size)});
   }
 
   Promise<void> write(ArrayPtr<const ArrayPtr<const byte>> pieces) override {
-    auto cork = writeBuffer.cork();
```

Comment:
Not sure if important, but FWIW it looks like the original code would only call `writeBuffer.cork()` for the multi-piece `write()` overload, but the new code would also call it for the single-buffer `write()` overload.